src/Sylius/Behat/Service/NotificationChecker.php

```diff
@@ -35,12 +35,21 @@ public function checkNotification(string $message, NotificationType $type): void
         foreach ($this->notificationAccessor->getMessageElements() as $messageElement) {
             if (
                 false !== strpos($messageElement->getText(), $message) &&
-                $messageElement->hasClass($type->__toString() === 'success' ? 'positive' : 'negative')
+                $messageElement->hasClass($this->resolveClass($type))
             ) {
                 return;
             }
         }
 
         throw new NotificationExpectationMismatchException($type, $message);
     }
+
+    private function resolveClass(NotificationType $type): string
+    {
+        if ($type->__toString() === 'info') {
+            return 'info';
+        }
+
+        return $type->__toString() === 'success' ? 'positive' : 'negative';
```

Comment:
What about having some type-class-map? Sth like:

```php
private function resolveClass(NotificationType $type): string
{
    $typeClassMap = [
        'info' => 'info',
        'failure' => 'negative',
        'success' => 'positive',
    ];

    Assert::keyExist($typeClassMap, $type->__toString());

    return $typeClassMap($type->__toString());
}
```

? I know it's a little bit longer, but much cleaner IMO and more extendable. But also it can be done as an improvement in a separate PR, so no pressure 🖖